src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs

```diff
@@ -430,9 +431,19 @@ private bool CanContinueBatch(ReplicationState state, ref long next)
                 }
             }
 
-            if (state.NumberOfItemsSent == 0)
+            if (state.ScannedItems == 1)
```

Comment:
Please check what happened if I have two large documents (each one is larger than `MaxSizeToSend`)
The first I need to skip, but the second I need to send.
If I get it correctly so, by the logic here we will never get to send the second one... please verify this.

Reply:
Maybe it would be simple to move the `CanContinueBatch` to the _end_ of the loop?
So we skip / update / add as needed, then break from the loop?

Reply:
the problem is, that we do not break tx boundaries, so need to fetch the next doc to check whether it is in the same tx or not

Reply:
good point

Reply:
If the allocated memory size is over the limit (after the first scan), we will break the tx because we will get to here:
https://github.com/ravendb/ravendb/blob/99c361de2aed22a7674a754c3a40b9d78c0ea264/src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs#L451